### ISSUE TITLE
Add support for Swift Package Manager to manage dependencies

### ISSUE THREAD

**Issue by akarsh** [cauliframework/cauli#199]
- [x] Add a section about swift package manager (spm) in brief similar to the pods to README.md
- [x] Add Package.swift
   - [x] modify other files to make SPM possible i.e. Application.swift
- [x] Follow the best practice mentioned in [Package Manager](https://swift.org/package-manager/)

**Comment by akarsh** [comment]
@brototyp 
If you agree with the above checklist, I would create a PR for adding support to SPM.


Can you add labels - "hacktoberfest" and "good first issue" to this issue?

I saw this PR: https://github.com/matomo-org/matomo-sdk-ios/pull/312/files for other SDK that was the inspiration to open this issue.

**Comment by pstued** [comment]
hey @akarsh 🙌
Thanks for contributing. Your issue looks good to me. Unless @brototyp has any concerns I'm looking forward to your Pull Request.

I was wondering, is there the need for us to create those labels? I think your contribution will count regardless of those labels for Hacktoberfest.

Happy Coding

**Comment by brototyp** [comment]
Hi @akarsh,

Sounds awesome! Maybe add a test that the sdk builds via SPM. For inspiration you can have a look [here](https://github.com/matomo-org/matomo-sdk-ios/pull/312/files#diff-52c976fc38ed2b4e3b1192f8a8e24cff).

I just double checked, the labels aren't necessary go be counted towards the t-shirt.

Good luck and please come back to us if there are any unforeseen issues. 😊

**Comment by akarsh** [comment]
Thank you @pstued  and @brototyp  :)

I managed to make a PR #200 

I was confused at first as i had multiple failures as the folder and project name was a bit different e.g. Cauli and project file - Cauliframework.xc.... But it was good learning I am able to understand the different parameters in the Package.swift file.

When I run with 
```
swift build
```
I got errors that UIKit module is missing. Then I tried with  this one
```
swift build \
        -Xswiftc "-sdk" -Xswiftc "`xcrun --sdk iphonesimulator --show-sdk-path`" \
        -Xswiftc "-target" -Xswiftc "x86_64-apple-ios12.0-simulator"
```
It is working. But I did not change the rake test command.
@brototyp should the rake test command for checking SPM to be changed or does it works fine with 
```
swift build
```
